cmds/boot/boot/boot.go

```diff
@@ -439,3 +256,99 @@ func main() {
 		log.Fatal(err)
 	}
 }
+
+// use syslinux parser
+
+func probeIsolinuxFiles() []string {
+	files := make([]string, 0, 10)
+	// search order from the syslinux wiki
+	// http://wiki.syslinux.org/wiki/index.php?title=Config
+	// TODO: do we want to handle extlinux too ?
+	dirs := []string{
+		"boot/isolinux",
+		"isolinux",
+		"boot/syslinux",
+		"syslinux",
+		"",
+	}
+	confs := []string{
+		"isolinux.cfg",
+		"syslinux.cfg",
+	}
+	for _, dir := range dirs {
+		for _, conf := range confs {
+			if dir == "" {
+				files = append(files, conf)
+			} else {
+				files = append(files, path.Join(dir, conf))
+			}
+		}
+	}
+	return files
+}
+
+func IsolinuxParseConfigWithSchemes(workingDir *url.URL, s curl.Schemes) (*syslinux.Config, error) {
+	for _, relname := range probeIsolinuxFiles() {
+		c, err := syslinux.ParseConfigFileWithSchemes(s, relname, workingDir)
+		if curl.IsURLError(err) {
+			continue
+		}
+		return c, err
+	}
+	return nil, fmt.Errorf("no valid syslinux config found")
+}
+
+// call IsolinuxBootImage(curl.DefaultSchemes, dir)
+func IsolinuxBootImage(schemes curl.Schemes, workingDir *url.URL) (*boot.LinuxImage, error) {
```

Comment:
should we put this in pkg/boot/syslinux?

Reply:
I've been wondering if we should get rid of all the "WithSchemes" variants and just always make users pass schemes. What do you think?

Reply:
hum... this code need to be refined to allow selecting the configuration as boot2 does or trying all configurations as systemboot/localboot does, so I was keeping it here as a test code for now.
Also the integration with mounting (mount, storage or diskboot package) is not there yet.
But I didn't consider the option to move it up to syslinux (and grub for the grub variant) package... :thinking:

Reply:
Yeah I'm hoping that eventually, all of them can return a slice of boot images sorted in the ranked order that each configuration presents. 

Like

Default entry
Followed by entries in the order they appear

Or whatever is appropriate for each config format.

Then, you'd call all the parsers, rank those, and present options to a user.

(For a different PR, of course, not here.)

Reply:
Thanks for sharing your vision !

Reply:
is it what you had in mind?

Reply:
Returning a slice is probably smarter that what I thought: I was probably going to return a structure with the map, as in grub a configuration can be referred by its name, label or position, and reference to submenu are in the form `1 > 2 > 3` to select the 3rd item in the second submenu of the first submenu ... but this is probably overkill, we might want to flatten to a slice, and maybe extend the attributes to store more than just a name. (It's nice though that you already added the name)

Reply:
Yeah the name thing comes from our internal shitty boot menu, which has been in my list of to-open-source.